src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java

```diff
@@ -115,6 +124,54 @@
         j.assertLogContains("ds.result=FAILURE", j.buildAndAssertSuccess(us));
     }
 
+    @Issue("JENKINS-60995")
+    @Test public void upstreamCause() throws Exception {
+        FreeStyleProject downstream = j.createFreeStyleProject("downstream");
+        WorkflowJob upstream = j.jenkins.createProject(WorkflowJob.class, "upstream");
+
+        upstream.setDefinition(new CpsFlowDefinition("build 'downstream'", true));
+        int numberOfUpstreamBuilds = 3;
+        for (int i = 0; i < numberOfUpstreamBuilds; i++) {
+            upstream.scheduleBuild2(0);
+            // Wait a bit before scheduling next one
+            Thread.sleep(100);
+        }
+        for (WorkflowRun upstreamRun : upstream.getBuilds()) {
+            j.waitForCompletion(upstreamRun);
+            j.assertBuildStatus(Result.SUCCESS, upstreamRun);
+        }
+        // Wait for all downstream builds to complete
+        j.waitUntilNoActivity();
+
+        List<BuildUpstreamCause> upstreamCauses = new ArrayList<>();
+        for (Run<FreeStyleProject, FreeStyleBuild> downstreamRun : downstream.getBuilds()) {
+            upstreamCauses.addAll(downstreamRun.getCauses().stream().filter(BuildUpstreamCause.class::isInstance).map(BuildUpstreamCause.class::cast).collect(Collectors.toList()));
```

Comment:
Readability nitpick:
```suggestion
            upstreamCauses.addAll(downstreamRun.getCauses().stream()
                .filter(BuildUpstreamCause.class::isInstance)
                .map(BuildUpstreamCause.class::cast)
                .collect(Collectors.toList()));
```